query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -136,12 +136,26 @@ final case class PeriodicSamplesMapper(startMs: Long,
     // time parameters
     offsetMs.map(o => rvs.map { rv =>
       new RangeVector {
-        val row = new TransientRow()
         override def key: RangeVectorKey = rv.key
         override def rows(): RangeVectorCursor = rv.rows.mapRow { r =>
```

Comment:
Sanity-check: probably we need to create a new row here? Given the current infrastructure, we can't update the row's timestamp directly?